samples/subsys/mgmt/updatehub/src/tls_config/user-tls-conf.h

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2023 O.S.Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define MBEDTLS_SHA256_C
+#define MBEDTLS_MD_C
```

Comment:
I'm not sure why you need this if FLASH_AREA_CHECK_INTEGRITY_MBEDTLS gets selected, that config would enable the mbedtls options. This seems like it would trump whatever mbedtls configuration a user has and only enable sha256 and mdx support, likely breaking their applications if they have their own custom file?